File: src/com/kevinlinxp/sublimeSnippetsSupport/SublimeSnippetsSupportConfigurable.java
package com.kevinlinxp.sublimeSnippetsSupport;

import com.intellij.openapi.options.BaseConfigurable;
import com.intellij.openapi.options.Configurable;
import com.intellij.openapi.options.ConfigurationException;
import com.intellij.openapi.options.SearchableConfigurable;
import com.intellij.openapi.util.Disposer;
import javax.swing.JComponent;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * @author Ke Lin
 * @version $Id$
 */
public class SublimeSnippetsSupportConfigurable extends BaseConfigurable implements
    SearchableConfigurable, Configurable.NoScroll {

  public final static String ID = "com.kevinlinxp.sublimeSnippetsSupport";

  private SublimeSnippetsSupportSettingsPanel myPanel;

  @NotNull
  @Override
  public String getId() {
    return ID;
  }

  @Nls
  @Override
  public String getDisplayName() {
    return "Sublime Snippets Support";
  }

  @Nullable
  @Override
  public JComponent createComponent() {
    myPanel = new SublimeSnippetsSupportSettingsPanel();
    return myPanel;
  }

  @Override
  public void apply() throws ConfigurationException {
    myPanel.apply();
  }

  @Override
  public void reset() {
    myPanel.reset();
  }

  public void disposeUIResources() {
    if (myPanel != null) {
      Disposer.dispose(myPanel);
    }
    myPanel = null;
  }
}

File: src/com/kevinlinxp/sublimeSnippetsSupport/SublimeSnippetsSupportConfigurable.kt
package com.kevinlinxp.sublimeSnippetsSupport

import com.intellij.openapi.options.BaseConfigurable
import com.intellij.openapi.options.Configurable
import com.intellij.openapi.options.ConfigurationException
import com.intellij.openapi.options.SearchableConfigurable
import com.intellij.openapi.util.Disposer
import javax.swing.JComponent

private class SublimeSnippetsSupportConfigurable : BaseConfigurable(), SearchableConfigurable, Configurable.NoScroll {

    companion object {
        const val ID = "com.kevinlinxp.sublimeSnippetsSupport"
    }

    private var myPanel: SublimeSnippetsSupportSettingsPanel? = null

    override fun getId(): String {
        return ID
    }

    override fun getDisplayName(): String {
        return "Sublime Snippets Support"
    }

    override fun createComponent(): JComponent? {
        myPanel = SublimeSnippetsSupportSettingsPanel()
        return myPanel
    }

    @Throws(ConfigurationException::class)
    override fun apply() {
        myPanel?.apply()
    }

    override fun isModified(): Boolean {
        return myPanel?.modified() ?: false
    }

    override fun reset() {
        myPanel?.reset()
    }

    override fun disposeUIResources() {
        if (myPanel != null) {
            Disposer.dispose(myPanel!!)
        }
        myPanel = null
    }
}
